torch/csrc/jit/codegen/cuda/scheduler/utils.cpp

```diff
@@ -1630,12 +1641,274 @@ void scheduleContiguousVectorLoad(
+  //  or a broadcast.
+  // Note: Currently don't really see a case where this function
+  //  should be called on a reduction output tv, but adding them
+  //  here for completeness.
+  std::deque<int> broadcast_or_reduction_pos;
+
+  // Map the root id's to their innermost concrete id's
+  //  on the leaf.
+  std::unordered_map<IterDomain*, int> root_id_to_leaf_pos;
+
+  // Move backward on tv's leaf domain;
+  for (auto i = ndims - 1; i >= 0; i--) {
+    auto leaf_id = tv->axis(i);
+    if (leaf_id->isBroadcast() || leaf_id->isReduction()) {
+      // Register this reduction or broadcast axis
+      //  to reorder.
+      broadcast_or_reduction_pos.push_front(i);
+      leftmost_pos = i;
+      continue;
+    }
+    auto maybe_root =
+        getMaybeRootIfInnermostTiled(leaf_id, maybe_rfactor_id_set);
+
+    if (maybe_root.has_value()) {
+      // Found an innermost id, add them to the
+      //  axes to reorder.
+      TORCH_INTERNAL_ASSERT(
+          root_id_to_leaf_pos.insert(std::make_pair(maybe_root.value(), i))
+              .second,
+          "Multiple \"innermost\" id seen for root id :",
+          maybe_root.value()->toString(),
+          " on ",
+          tv->toString(),
+          " very likely an invariant is broken.");
+      leftmost_pos = i;
+    } else {
+      break;
+    }
+  }
+
+  // Calculate the ordering:
+
+  // pointer to the current target postion after
+  //  repordering
+  int current_pos = leftmost_pos;
+  std::unordered_map<int, int> reorder_map;
+
+  // first place all the broadcast and reduction on the left:
+  for (auto original_broadcast_or_reduction_pos : broadcast_or_reduction_pos) {
+    reorder_map[original_broadcast_or_reduction_pos] = current_pos++;
+  }
+
+  // Next put all the innermost leaf id's
+  for (auto root_id : tv->getMaybeRFactorDomain()) {
+    auto leaf_id_pos_it = root_id_to_leaf_pos.find(root_id);
+    if (leaf_id_pos_it != root_id_to_leaf_pos.end()) {
+      reorder_map[leaf_id_pos_it->second] = current_pos++;
+    }
+  }
+
+  // Validate that we have processed all inner ids or broadcast/reduction
+  //  ids we have registered.
+  TORCH_INTERNAL_ASSERT(current_pos == ndims, "Inconsistent ordering logic");
+
+  // Apply the new order:
+  tv->reorder(reorder_map);
+}
+
+struct MatmulSelector : public MaxInfoSpanningTree::Selector {
+ public:
+  explicit MatmulSelector(std::unordered_set<TensorView*> selected_tvs)
+      : selected_tvs_(selected_tvs) {}
+
+  explicit MatmulSelector() : propagate_all_(true) {}
+
+  bool allowC2P(TensorView* from, TensorView* to) final {
+    return propagate_all_ || selected_tvs_.count(to);
+  }
+  bool allowP2C(TensorView* from, TensorView* to) final {
+    return propagate_all_ || selected_tvs_.count(to);
+  }
+  bool allowSibling(TensorView* from, TensorView* to) final {
+    return propagate_all_ || selected_tvs_.count(to);
+  }
+
+ private:
+  std::unordered_set<TensorView*> selected_tvs_;
+  bool propagate_all_ = false;
```

Comment:
Is this `propagate_all_` used? If you want to select all, you could just pass nullptr to the `selector` argument of the ctor of `MaxRootDomainInfoSpanningTree`:
https://github.com/csarofeen/pytorch/blob/942be5b256056d0e02877361b814ae6af32ca15f/torch/csrc/jit/codegen/cuda/maxinfo_propagator.h#L231-L233

Reply:
Oh didn't realize the default behavior without a selector would be `selectAll`. That make sense. Updated the logic to use that. Thanks!